File: BotPackages/Typescript/api/do-turn.ts
import { TurnStateDto } from "../types/turn-state.dto";
import { Move, TurnDecision } from "../types/turn-decision.dto";
import simpleBot from "../example-bots/simple-bot";

export default function doTurn(turnState: TurnStateDto): TurnDecision {
  // return simpleBot.doTurn(turnState);

  return {
    droneDecisions: turnState.drones.map((drone, index) => ({
      id: drone.id,
      move:
        drone.holding == null &&
        turnState.packages.some((pkg) => pkg.x == drone.x && pkg.y == drone.y)
          ? Move.PickOrDrop
          : ((Math.floor(Math.random() * 4) + 1) as Move),
    })),
  };
}

File: BotPackages/Python-Flask/api/do_turn.py
import random
from model.TurnState import DroneState, PackageState, TurnState
from model.TurnDecision import DroneDecision, Move, TurnDecision


def do_turn(turn_state: TurnState) -> TurnDecision:
    return TurnDecision(
        droneDecisions=list(
            map(
                lambda drone: DroneDecision(
                    id=drone.id,
                    move=Move.PickOrDrop
                    if drone.holding == None
                    and any(package_under_drone(p, drone) for p in turn_state.packages)
                    else random.randrange(4) + 1,
                ),
                turn_state.drones,
            )
        )
    )


def package_under_drone(pkg: PackageState, drone: DroneState):
    return pkg.x == drone.x and pkg.y == drone.y
